frontend/hooks/useBuilds.js

```diff
@@ -0,0 +1,60 @@
+import { useQuery, useQueryClient, useMutation } from '@tanstack/react-query';
+import api from '@util/federalistApi';
+
+// Assumes the order of builds is based on createdAt desc
+export function getLatestBuildByBranch(builds) {
+  const latestBranches = [];
+
+  return builds.map((build) => {
+    if (build.completedAt && !latestBranches.includes(build.branch)) {
+      latestBranches.push(build.branch);
+      return { ...build, latestForBranch: true };
+    }
+
+    return { ...build, latestForBranch: false };
+  });
+}
+
+export function useBuilds(siteId) {
+  const { isPending, error, data } = useQuery({
+    queryKey: ['builds', parseInt(siteId, 10)],
+    queryFn: () =>
+      api.fetchBuilds({ id: siteId }).then((builds) => getLatestBuildByBranch(builds)),
+    refetchInterval: 10000,
```

Comment:
**non-blocking**: ideally this would be a constant, even if it's just defined at the top of the file

Reply:
Created a `utils` file in the hooks. Not sure if we like that there.